src/gtc/passes/oir_optimizations/utils.py

```diff
@@ -217,16 +219,17 @@ def new_symbol_name(name: str) -> str:
     return new_symbol_name
 
 
-def collect_symbol_names(node: BaseNode) -> Set[str]:
+def collect_symbol_names(node: eve.RootNode) -> Set[str]:
```

Comment:
Not really related to this PR, but I would like to check how this function relates to `eve.traits.SymbolsCollector` because I think they have overlapping functionality but there are some differences